OpenRA.Game/Network/OrderIO.cs

```diff
@@ -109,16 +120,32 @@ public static bool TryParseSync(byte[] packet, out (int Frame, int SyncHash, ulo
 			return true;
 		}
 
-		public static bool TryParseAck((int FromClient, byte[] Data) packet, out int frame)
+		public static bool TryParseAck((int FromClient, byte[] Data) packet, out int frame, out int count)
 		{
 			// Ack packets are only accepted from the server
-			if (packet.FromClient != 0 || packet.Data.Length != 5 || packet.Data[4] != (byte)OrderType.Ack)
```

Comment:
I expect that this length check removal is the cause of the IndexOutOfRangeException.